### PR TITLE
docs: update README to be auto-included in openfeature.dev

### PR DESCRIPTION
**Pull request by toddbaert**
Updating README to conform to auto-inclusion for openfeature.dev

### CODE REVIEW

File: README.md
Comment (beeme1mr) [inline README.md:28]:
```suggestion
  <a href="https://bestpractices.coreinfrastructure.org/projects/6241">
    <img alt="CII Best Practices" src="https://bestpractices.coreinfrastructure.org/projects/6241/badge" />
```

File: README.md
Comment (beeme1mr) [inline README.md:30]:
```suggestion
  <a href="https://maven-badges.herokuapp.com/maven-central/dev.openfeature/sdk">
```

**Review by beeme1mr — APPROVED** [review]
Looks good! Since you're referencing the in-memory provider, we should wait until the next release is ready to be merged.

https://github.com/open-feature/java-sdk/pull/558

File: README.md
Comment (Kavindu-Dodan) [inline README.md:201]: To align with all other examples, 

```suggestion
```java
  // add a hook globally, to run on all evaluations
  OpenFeatureAPI api = OpenFeatureAPI.getInstance();
  api.addHooks(new ExampleHook());
  
  // add a hook on this client, to run on all evaluations made by this client
  Client client = api.getClient();        
  client.addHooks(new ExampleHook());
  
  // add a hook for this evaluation only
  Boolean retval = client.getBooleanValue(flagKey, false, null,
          FlagEvaluationOptions.builder().hook(new ExampleHook()).build());
```

**Review by Kavindu-Dodan — APPROVED** [review]
Left two comments, but the rest looks solid

File: README.md
Comment (toddbaert) [reply README.md:319]: I was debating that. If you think it's a good idea too, I'll make that change.

**Pull request merged**
